sqlite/src/main.c

```diff
@@ -3433,13 +3433,15 @@ static int openDatabase(
 #endif
 
 #if defined(SQLITE_BUILDING_FOR_COMDB2)
-  static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
-  pthread_mutex_lock(&mutex);
-  /* these modify global structures */
-  register_lua_sfuncs(db, thd);
-  register_lua_afuncs(db, thd);
-  register_date_functions(db); 
-  pthread_mutex_unlock(&mutex);
+  if( thd!=NULL ){
```

Comment:
Fix for NULL "thd", part 2.  The registered Lua functions assume "thd" cannot be NULL; therefore, they cannot be added when "thd" is in fact NULL.